### ISSUE TITLE
[PS Vita] Attempting to use a virtual Note button (using Macrofire) does nothing.

### ISSUE THREAD

**Issue by rhobiusvoid** [VanLaser/analog-controls-for-armored-core-ps2#6]
Using Macrofire (and a bit of .txt editing) does not work for the Purge button modifiers, such as replacing the L button (used for boosting in my case) to be a NOTE button. Works as intended on PPSSPP and a real PSP, but unsure what may be the cause of this, if it is related to the plugin or how the code is searching for the Note button.

**Comment by VanLaser** [comment]
Indeed, all the testing was done on PPPSSPP (I don't have a Vita). Apparently it's a PS Vita issue (from the cdromance "Armored Core 3 Portable - True Analogs Mod (USA)" disqus comments thread): 

![ps-vita-comment](https://github.com/VanLaser/analog-controls-for-armored-core-ps2/assets/8756008/b55b0cf7-9a7a-473f-b7f8-2b0feeae27d4)

But if that Macrofire thing can inject that Note button properly, the purge should still work. All I can say is the mod uses the "right" value sent by the Note button inside the game to enable the extra purge modifier. Sorry I can't help more there.

**Comment by VanLaser** [comment]
If that Macrofire can fire multiple buttons at once, you could perhaps make it send the "classic" 4 combo keys that are shown in the in-game keymap as purge modifier when you press L instead?

**Comment by rhobiusvoid** [comment]
That could be an idea, but that's mostly figuring out which buttons I actually need to send, especially due to custom key assignments. Also hey, I'm that same dude from the Disqus thread. Macrofire works for replacing the button (since L no longer boosts and instead just does nothing in-game) but that just leaves me stumped regarding _why_ the game just doesn't recognize Note.

Knowing Adrenaline, it likely never actually sends a Note input because why would it when it doesn't exist on the Vita, heh.

**Comment by VanLaser** [comment]
The purge combo buttons should be those shown by the in-game keymap customization menu, when you scroll down there it shows you all the purge combinations available (or "Disabled" in same cases when you remap things). One idea maybe? would be to revert to the default in-game keymap and remap things (including firing multiple buttons for purging) from outside the game.

**Comment by rhobiusvoid** [comment]
What I did back then on the PSP (before upgrading to Vita) was just use the default control scheme and remap to my usual custom scheme using Macrofire. That way, I could still purge while massively improving the controls (at least personally). Nowadays I just play the PS2 versions anytime I want (good work on the pre-Nexus stuff by the way) but it could be nice to pop open ACLR on my Vita from time to time.

**Comment by VanLaser** [comment]
Thanks :) Right, that kind of custom remapping is  what I'm suggesting also for the Vita, if possible in any way (?)

**Comment by rhobiusvoid** [comment]
As for any remapping on PS Vita, there's only reVita (as the best and latest one) alongside the capability to pair a DS3/4 controller to the PS Vita (using PSTV backported protocols iirc). Could possibly change that Note button to be an L2 button, and then use reVita to bind a part of the touchscreen to L2.
![image](https://github.com/VanLaser/analog-controls-for-armored-core-ps2/assets/22803374/c329c807-1279-48bc-8b99-14716ea16b31)

**Comment by VanLaser** [comment]
Also if mapping to "Note" doesn't work, if you can map one button to fire up 4 of them (L1+R1+square+triangle by default), that can also remake the original purge modifier combo.

**Comment by rhobiusvoid** [comment]
This is personally what I used previously in Macrofire, for the default control scheme. I'd imagine that if I wanted to do this in custom control schemes, I'd just need to change the remapped combos to match with the new buttons.
[last raven.txt](https://github.com/VanLaser/analog-controls-for-armored-core-ps2/files/13762485/last.raven.txt)
